packages/webapp/src/components/DateRangeSelector/CustomDateRangeSelector.jsx

```diff
@@ -33,10 +33,25 @@ export default function CustomDateRangeSelector({
   changeEndDate,
   fromDateMax,
   toDateMin,
+  resetDateRangeOption,
 }) {
   const { t } = useTranslation();
   const inputStyle = { classes: { container: { minWidth: 'calc((100% - 24px) / 2)' } } };
 
+  const isValidRef = useRef(isValid);
```

Comment:
Ohhh, you're right, this won't work without a ref either. I do think we should try to find a solution here not involving refs, since refs are usually meant to be used as a last resource kind of thing because they step out of the "normal" React workflow, so in general they're used for logic that won't affect rendering which isn't the case here.

I'm curious about what wasn't working with your first approach where you checked `isValid` on the `onClickAway` method?

```js
const onClickAway = () => {
    if (!isCustomDatePickerOpen) {
      return;
    }
    if (!isValid) {
      setSelectedDateRangeOption(options[0]);
    }
    setIsCustomDatePickerOpen(false);
  };
```

That seems to be working well in my testing but I could be missing something. Alternatively we could also setup an effect like so

```js
useEffect(() => {
  if (!isValid & !isCustomDatePickerOpen) {
    setSelectedDateRangeOption(options[0]);
  }
}, []);
```

Reply:
In addition to clicking away, the picker can be closed by clicking the selector as you can see in Duncan's first screen recording [here](https://github.com/LiteFarmOrg/LiteFarm/pull/2913#issuecomment-1775520157) (0:55~).
That was what I had to handle also, and I used a ref.

Reply:
Ahhh gotcha, thank you for clarifying! Would the `useEffect` alternative work for that scenario? It should actually be something like

```js
useEffect(() => {
  if (!isValid & !isCustomDatePickerOpen) {
    setSelectedDateRangeOption(options[0]);
  }
}, [isValid, isCustomDatePickerOpen]);
```

I had missed the dependencies on the other comment!

Reply:
It didn't work with the validation that I had, but I removed it, so it should be good. The validation was like an insurance, so we should not have a problem without it.